scargo/scargo_src/sc_flash.py

```diff
@@ -57,10 +57,19 @@ def flash_stm32(config: Config, flash_profile: str = "Debug") -> None:
     logger = get_logger()
 
     project_path = get_project_root()
-    bin_name = config.project.bin_name.lower()
+    bin_name = config.project.bin_name
+    if not bin_name:
+        logger.error("No bin_name in config!")
+        return
+    bin_name = bin_name.lower()
     bin_path = project_path / "build" / flash_profile / "bin" / f"{bin_name}.bin"
 
-    flash_start = hex(config.stm32.flash_start)
+    stm32_config = config.stm32
```

Comment:
Shouldn't this fail while parsing toml if target is stm32?

Reply:
Maybe, but mypy will still not understand it, so it's outside the scope here

Reply:
I'd rather write exception for this and make sure it's handled correctly in sc_config. What do you think?

Reply:
not sure what you have in mind, but I may add a method like `Config.get_stm32_config()` as I've mentioned in another comment